### DISCUSSION TITLE
Prevent negative weights in ordinary kriging

### DISCUSSION THREAD

**Discussion by diedro2** [GeoStat-Framework/PyKrige#207] [Ideas]
Dear all,
I would like to know if PyKrige is able to deal with negative weights which could arise with the ordinary kriging. An example is provided by DEUTSCH et al. Is this problem overcome by the weight option where "the weights are calculated from a logistic function, so weights at small lags are ~1 and weights at the longest lags are ~0"?
Thanks in advance

**Comment by MuellerSeb** [comment]
Hey there,

I don't really understand your questions. The weights option you mentioned is used for variogram fitting, so the values at smaller lags have more weight during fitting the theoretical variogarm model.

Could you specify which weights you mean? Or what the concrete example of ***Deutsch et al.*** is?

Sebastian

**Reply by diedro2** [reply]
As far I have understood, when I want to compute the unknown values of my computation domain near points where values are known the kriging weights resulting from the kriging linear system could results negative.  Here an example: https://www.sciencedirect.com/science/article/pii/0098300496000052

**Reply by MuellerSeb** [reply]
That is really interesting. I wasn't aware that it could be undesired that these weight get out of the intervall [0,1]. What you basically do with this restriction is to not leave the interval of maximum and minimum of the input data. Problem is, that you need to store the weights and check and manipulate them for each output point. That is not implemented at the moment. I guess we could add this as an idea for PyKrige2.

**Reply by diedro2** [reply]
Yes, you are right. I have implemented it in one of my old FORTRAN  codes. Are you planning to develop a PyKrige2? Would do you like a piece of my FORTRAN code?

**Reply by MuellerSeb** [reply]
I guess it is straight forward to implement. Let's see if we want to add this. But thanks for that hint! :+1:
